### ISSUE TITLE
H.264 Video Decoding Artifacts with AMD GPU Decoding

### ISSUE THREAD

**Issue by Gryzzplo** [clsid2/mpc-hc#1429]
Video artifacts appearing when watching specific H.264 Blu-ray rips. The ones affected in my collection are WALL-E, Up, and F&F Tokyo Drift. All of which were all untouched video streams from the Blu-ray disc.

Link to the video affected (97MB): https://bit.ly/3I5fwoE

System Info:
Ryzen 9 5900HX 
6800M
Both the 5900HX iGPU and 6800M caused artifacts with DXVA2 and D3D11.

I also tried playing this back using an Intel iGPU and a Laptop 3060 (DXVA2 enabled on the 3060, not on the iGPU) and both had no issues. The issue seems to be limited to AMD only.
<img width="1280" alt="MPC-HC Artifacts Hardware Decoding" src="https://user-images.githubusercontent.com/95281886/144025572-576a4e76-8d72-44c2-a4ca-423a04729d21.png">
.

**Comment by clsid2** [comment]
Report driver bug to AMD:
https://www.amd.com/en/support/kb/faq/amdbrt

**Comment by xLn2** [comment]
In my previos experiment I had seen something similar like this big green pixelarated video. I noticed Nvidia color settings was edited so I changed settings to default and problem disappeared. I may suggest you to reinstall graphic driver with factory reset option.

**Comment by Gryzzplo** [comment]
The thing is this is a laptop (sorry if I didn't make that clear, it's an ASUS G15 AE) and I've tried uninstalling and installing graphics drivers from both ASUS and AMD and none of that seemed to fix the issue, and yes I did tick the factory reset option. I also tried lowering the screen's refresh rate, resolution from 1440p to 1080p and no luck. I've already submitted a bug report to AMD via their software and hopefully they'll address it.

**Comment by clsid2** [comment]
I suggest making another report to AMD.

This time include the screenshot and the sample video link in the bug report. Also mention that it only happens with certain videos, and that those same videos work fine with Intel and NVIDIA. The sample file is important as that can help AMD to more easily reproduce the problem, which is essential for increasing the chance of a quick fix.

**Comment by Gryzzplo** [comment]
Yes, I've already included the screenshot included in this post as well as another screenshot with hardware decoding disabled and the link to the video in the report and here's what I wrote in the description (I basically copied the desc. from the report here):

Video artifacts appearing when watching specific H.264 Blu-ray rips. The ones affected in my collection are WALL-E, Up, and F&F Tokyo Drift. All of which were all untouched video streams from the Blu-ray disc

and here's what I wrote in the steps to reproduce section:

1. Download the video affected via the included link on OneDrive (99MB)
2. Open MPC-HC or any media player with hardware decoding (DXV2 or D3D11) enabled
3. Play the downloaded video file.
4. Observe the artifacts in the video player.
Note that the video file is not corrupted and the issue is not experienced when using software decoding or hardware decoding with an Intel or NVIDIA GPU, the 2nd link provided shows this with hardware decoding disabled.
